src/python/gudhi/simplex_tree_multi.pyx

```diff
@@ -0,0 +1,970 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Vincent Rouvreau
+#
+# Copyright (C) 2016 Inria
+#
+# Modification(s):
+#	- 2022/11 Hannah Schreiber / David Loiseaux : adapt for multipersistence. 
+#   - YYYY/MM Author: Description of the modification
+
+
+
+__author__ = "Vincent Rouvreau"
+__copyright__ = "Copyright (C) 2016 Inria"
+__license__ = "MIT"
+
+
+from cython.operator import dereference, preincrement
+from libc.stdint cimport intptr_t
+from libc.stdint cimport uintptr_t
+from typing import Any
+
+cimport numpy as cnp
+import numpy as np
+cnp.import_array()
+
+cimport gudhi.simplex_tree_multi
+cimport cython
+from gudhi import SimplexTree ## Small hack for typing
+
+from filtration_domination import remove_strongly_filtration_dominated, remove_filtration_dominated
+
+
+from warnings import warn
+
+ctypedef double value_type
+
+cdef extern from "Simplex_tree_multi.h" namespace "Gudhi":
+	void multify(const uintptr_t, const uintptr_t, const unsigned int) nogil
+	void flatten(const uintptr_t, const uintptr_t, const unsigned int) nogil
+	void flatten_diag(const uintptr_t, const uintptr_t, const vector[value_type], int) nogil
+	void squeeze_filtration(uintptr_t, const vector[vector[value_type]]&, bool) nogil
+	vector[vector[value_type]] get_filtration_values(uintptr_t) nogil
+
+
+cdef bool callback(vector[int] simplex, void *blocker_func):
+	return (<object>blocker_func)(simplex)
+
+# SimplexTree python interface
+cdef class SimplexTreeMulti:
```

Comment:
Do we want to encode in the name the fact that it is restricted to 1-critical filtrations, or is it unnecessary?

Reply:
This class supports multi-critical filtrations. Knowing the number of parameter `n`, the multiple "birthtimes" of a simplex filtration `f`, can be recovered using, e.g., `np.array_split(f, len(f) // n)`.  c.f. the attached notebook on the initial message.
It is true, though, that a lot of functions do not support multi-critical filtrations yet; they should all have a comment saying that.

Reply:
Ah, ok, I misinterpreted those comments.